ios/private/common_utils.bzl

```diff
@@ -0,0 +1,172 @@
+    hasUITests: Whether or not to generate ios_ui_test tests
+    needsXCTest: set the 'testonly' attribute on swift_library
+    bundle_name: optionally override the name used for the resource bundle
+  """
+
+  # if we are backed by a JS package, these attributes
+  # will be populated to add to the sources/resources of the
+  # swift_library
+  data = []
+  resourceSources = []
+
+  bundleName = bundle_name if bundle_name != None else name
+
+  if len(resources) > 0:
+    apple_resource_bundle(
+      name = name + "ResourceBundle",
+      bundle_name = bundleName,
+      bundle_id = "com.intuit.ios.player.resources."+name,
+      resources = resources,
+    )
+
+    ios_bundle_module_shim(bundleName)
+    data.append(":" + name + "ResourceBundle")
+    resourceSources.append(":" + bundleName + "ResourceShim")
+
+  # Group up files to be used in swift_library
+  # and in //:PlayerUI_Pod which builds the zip of sources
+  pkg_files(
+    name = name + "_Sources",
+    srcs = native.glob(["Sources/**/*.swift"]),
+    strip_prefix = strip_prefix.from_pkg(),
+    visibility = ["//visibility:public"],
+  )
+
+  swift_library(
+      name = name,
+      module_name = name,
+      srcs = [":" + name + "_Sources"] + resourceSources,
+      visibility = ["//visibility:public"],
+      testonly = needsXCTest,
+      deps = deps,
+      data = data,
+      # this define makes Bundle.module extension work from ios_bundle_module_shim
+      defines = ["BAZEL_TARGET"]
+  )
+
+  # Packages not specific to SwiftUI don't need ViewInspector
+  # so it can just be regular unit tests
+  if hasUnitTests == True:
+    ios_unit_test(
+        name = name + "Tests",
+        srcs = native.glob(["Tests/**/*.swift"]),
+        minimum_os_version = "14.0",
+        deps = [
+          ":" + name
+        ] + deps + test_deps,
+        visibility = ["//visibility:public"]
+    )
+  # ViewInspector has to run as a UI Test to work properly
+  # SwiftUI plugins need ViewInspector
+  if hasViewInspectorTests == True:
+    ios_ui_test(
+        name = name + "ViewInspectorTests",
+        srcs = native.glob(["ViewInspector/**/*.swift"]),
+        minimum_os_version = "14.0",
+        deps = [
+            "@swiftpkg_viewinspector//:Sources_ViewInspector",
+            ":" + name
+        ] + deps + test_deps,
+        visibility = ["//visibility:public"],
+        test_host = test_host
+    )
+    
+  # Runs SwiftLint as a test calling the genrule target which outputs the result of linting
+  native.sh_test(
+    name = name + "SwiftLint",
+    srcs = [":"+ name + "_Lint"],
+    visibility = ["//visibility:public"],
+  )
+
+  # Runs the SwiftLint as part of the build, if lint fails with serious violations defer the results for the test
+  native.genrule(
+    name = name + "_Lint",
+    tools = [
+      "@SwiftLint//:swiftlint"
+    ],
+    srcs = [":" + name + "_Sources"] + ["//:.swiftlint.yml"],
+    outs = ["output.sh"],
+    executable = True,
```

Comment:
can you also add `testonly = True`

this will ensure it can only be depended on by test type rules, to prevent accidental usage